python/python/glide/async_commands/sorted_set.py

```diff
@@ -0,0 +1,154 @@
+# Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
+
+from enum import Enum
+from typing import List, Optional, Union
+
+
+class InfBound(Enum):
+    """
+    Enumeration representing numeric and lexicographic positive and negative infinity bounds for sorted set scores.
+    """
+
+    POS_INF = {"default_arg": "+inf", "lex_arg": "+"}
+    NEG_INF = {"default_arg": "-inf", "lex_arg": "-"}
+
+
+class ScoreBoundary:
+    """
+    Represents a specific numeric score boundary in a sorted set.
+
+    Args:
+        value (float): The score value.
+        is_inclusive (bool): Whether the score value is inclusive. Defaults to True.
+    """
+
+    def __init__(self, value: float, is_inclusive: bool = True):
+        # Convert the score boundary to the Redis protocol format
+        self.value = str(value) if is_inclusive else f"({value}"
+
+
+class LexBoundary:
+    """
+    Represents a specific lexicographic boundary in a sorted set.
+
+    Args:
+        value (str): The lex value.
+        is_inclusive (bool): Whether the score value is inclusive. Defaults to True.
+    """
+
+    def __init__(self, value: str, is_inclusive: bool = True):
+        # Convert the lexicographic boundary to the Redis protocol format
+        self.value = f"[{value}" if is_inclusive else f"({value}"
+
+
+class Limit:
+    """
+    Represents a limit argument for a range query in a sorted set to be used in [ZRANGE](https://redis.io/commands/zrange) command.
+
+    The optional LIMIT argument can be used to obtain a sub-range from the matching elements
+        (similar to SELECT LIMIT offset, count in SQL).
+    Args:
+        offset (int): The offset from the start of the range.
+        count (int): The number of elements to include in the range.
+            A negative count returns all elements from the offset.
+    """
+
+    def __init__(self, offset: int, count: int):
+        self.offset = offset
+        self.count = count
+
+
+class RangeByIndex:
+    """
+    Represents a range by index (rank) in a sorted set.
+
+    The `start` and `stop` arguments represent zero-based indexes.
+
+    Args:
+        start (int): The start index of the range.
+        stop (int): The stop index of the range.
+    """
+
+    def __init__(self, start: int, stop: int):
+        self.start = start
+        self.stop = stop
+
+
+class RangeByScore:
+    """
+    Represents a range by score in a sorted set.
+
+    The `start` and `stop` arguments represent score boundaries.
+
+    Args:
+        start (Union[InfBound, ScoreBoundary]): The start score boundary.
+        stop (Union[InfBound, ScoreBoundary]): The stop score boundary.
+        limit (Optional[Limit]): The limit argument for a range query. Defaults to None.
```

Comment:
Lets refer to the Limit class for more info, try to find python docstring way of doing it (maybe https://stackoverflow.com/questions/21289806/link-to-class-method-in-python-docstring)